plugins/modules/pnp_intent.py

```diff
@@ -763,49 +879,47 @@ def get_have(self):
                 function='get_software_image_details',
                 params=self.want.get("image_params"),
             )
-
-            self.log(str(image_response))
-
             image_list = image_response.get("response")
+            self.log(str(image_response))
 
-            if len(image_list) == 1:
-                have["image_id"] = image_list[0].get("imageUuid")
-                self.log("Image Id: " + str(have["image_id"]))
-            else:
-                self.module.fail_json(msg="Image not found", response=[])
-
-            # check if given template exists, if exists store template id
+            # check if project has templates or not
             template_list = self.dnac_apply['exec'](
                 family="configuration_templates",
                 function='gets_the_templates_available',
                 params={"project_names": self.want.get("project_name")},
             )
-
             self.log(str(template_list))
 
-            if template_list and isinstance(template_list, list):
-                # API execution error returns a dict
-                template_details = get_dict_result(template_list,
-                                                   'name', self.want.get("template_name"))
-                if template_details:
-                    have["template_id"] = template_details.get("templateId")
-
-                    self.log("Template Id: " + str(have["template_id"]))
-                else:
-                    self.module.fail_json(msg="Template not found", response=[])
-            else:
-                self.module.fail_json(msg="Project Not Found", response=[])
-
             # check if given site exits, if exists store current site info
-            site_name = self.want.get("site_name")
-
             site_exists = False
-            (site_exists, site_id) = self.site_exists()
+            if isinstance(self.want.get("site_name"), str):
+                site_name = self.want.get("site_name")
+                (site_exists, site_id) = self.site_exists()
 
             if site_exists:
                 have["site_id"] = site_id
                 self.log("Site Exists: " + str(site_exists) + "\n Site_id:" + str(site_id))
                 self.log("Site Name:" + str(site_name))
+                if self.want.get("pnp_type") == "AccessPoint":
+                    if self.get_site_type() != "floor":
+                        self.module.fail_json(msg="Type of the site must \
+                            be a floor for claiming an AP", response=[])
+                if len(image_list) == 1:
+                    have["image_id"] = image_list[0].get("imageUuid")
+                    self.log("Image Id: " + str(have["image_id"]))
+                if template_list and isinstance(template_list, list):
+                    # API execution error returns a dict
+                    if self.want.get("template_name"):
```

Comment:
we are getting the template_name in next line as well. Instead, can we store and use it?

```
    if template_list and isinstance(template_list, list):
            template_name = self.want.get("template_name")
            template_details = get_dict_result(template_list, 'name', template_name)

            if template_details:
                have["template_id"] = template_details.get("templateId")
            else:
                self.module.fail_json(msg="Template Not Found", response=[])
        else:
            self.module.fail_json(msg="Project Not ......."
```